src/Mod/TechDraw/App/DrawPagePyImp.cpp

```diff
@@ -28,100 +49,91 @@ std::string DrawPagePy::representation() const
 
 PyObject* DrawPagePy::addView(PyObject* args)
 {
-    //this implements iRC = pyPage.addView(pyView)  -or-
-    //doCommand(Doc, "App.activeDocument().%s.addView(App.activeDocument().%s)", PageName.c_str(), FeatName.c_str());
     PyObject *pcDocObj;
-
-    if (!PyArg_ParseTuple(args, "O!", &(App::DocumentObjectPy::Type), &pcDocObj)) {
-        PyErr_SetString(PyExc_TypeError, "DrawPagePy::AddView - Bad Arg - not DocumentObject");
+    if (!PyArg_ParseTuple(args, "O!", &(TechDraw::DrawViewPy::Type), &pcDocObj)) {
```

Comment:
Re failure in DrawHatchTest.py: Despite the later comment, addView has to accept DocumentObjects as DrawHatch and DrawGeomHatch are not DrawView, but are DocumentObject.

Reply:
Ok, but that cast to `DrawView` is undefined behaviour.
Why not make `DrawHatch` and `DrawGeomHatch` inherit from `DrawView`?

Reply:
Ok, but that cast to DrawView is undefined behaviour.
Why not make DrawHatch and DrawGeomHatch inherit from DrawView?

Reply:
I guess it was because they have no independent existence.   If they were being created today, they would probably be extensions of DrawViewPart.

I'm wrong on my comment though.  The test case shouldn't be adding the hatch to the page.  They belong only to DrawViewPart & derived classes.  I'll make a change to the test case.